test/extensions/filters/listener/tls_inspector/tls_utility.cc

```diff
@@ -40,6 +41,146 @@ std::vector<uint8_t> generateClientHello(uint16_t tls_min_version, uint16_t tls_
   return buf;
 }
 
+std::vector<uint8_t> generateClientHelloFromJA3Fingerprint(const std::string& ja3_fingerprint) {
+  // fingerprint should have this format:
+  //  SSLVersion,Cipher,SSLExtension,EllipticCurve,EllipticCurvePointFormat
+  // Example:
+  //   769,47-53-5-10-49161-49162-49171-49172-50-56-19-4,0-10-11,23-24-25,0
+  std::vector<std::string> fingerprint = absl::StrSplit(ja3_fingerprint, ',');
+  if (fingerprint.size() != 5) {
```

Comment:
Since this is test code, I think ASSERT(fingerprint.size() == 5) is better than an empty return.